examples/dixonsolve.C

```diff
@@ -0,0 +1,286 @@
+typedef DenseVector<Ints> ZVector;
+
+
+struct FixPrime {
+    typedef Givaro::Integer Prime_Type;
+    const Prime_Type _myprime;
+    FixPrime(const Givaro::Integer& i) : _myprime(i) {}
+    inline FixPrime &operator ++ () { return *this; }
+    const Prime_Type &operator * () const { return randomPrime(); }
+    const Prime_Type & randomPrime() const { return _myprime; }
+    void setBits(uint64_t bits) {}
+    template<class _ModField> void setBitsField() { }
+};
+
+template<typename _Matrix, typename _EliminationMethod>
+int test(_Matrix A, std::string vector_file, bool inv, bool pp, bool sparse_elim) {
+    // TODO : git rm dixondenseLU dixonsparseelim
+
+    std::cout << "A is " << A.rowdim() << " by " << A.coldim() << std::endl;
+
+    if (pp)
+    {
+            // Print Matrix
+
+            // Matrix Market
+            // std::cout << "A is " << A << std::endl;
+
+            // Maple
+        A.write(std::cout << "A:=", Tag::FileFormat::Maple) << ';' << std::endl;
+    }
+
+        // Vector File
+    Ints ZZ;
+    std::ifstream invect;
+
+    ZVector B(ZZ, A.rowdim());
+
+    bool createB = vector_file.empty();
+    if (!createB) {
+        invect.open (vector_file, std::ifstream::in);
+        if (!invect) {
+            createB = true;
+        } else {
+            for(ZVector::iterator it=B.begin(); it != B.end(); ++it)
+                invect >> *it;
+        }
+    }
+
+        // Vectors
+    ZVector X(ZZ, A.coldim());
+
+    if (createB) {
+        ZVector U(ZZ, A.coldim());
+        if (inv) {
+            std::cerr << "Creating a random {-1,1} vector " << std::endl;
+            srand48( BaseTimer::seed() );
+            for(auto& it:B)
+                if (drand48() <0.5)
+                    it = -1;
+                else
+                    it = 1;
+        } else {
+            std::cerr << "Creating a random consistant {-1,1} vector " << std::endl;
+            srand48( BaseTimer::seed() );
+            for(FFPACK::rns_double::integer& it:U)
+                if (drand48() <0.5)
+                    it = -1;
+                else
+                    it = 1;
+
+            // B = A U
+            A.apply(B,U);
+        }
+    }
+
+    if(pp)
+    {
+            // Print RHS
+        B.write(std::cout << "B:=", Tag::FileFormat::Maple) << ';' << std::endl;
+    }
+
+    std::cout << "B is " << B.size() << "x1" << std::endl;
+
+    Timer chrono;
+
+        // BlasElimination
+    // TODO : à vérifier si cela marche avec Method::DenseElimination
+    _EliminationMethod M;
+    if (inv){
+        M.singularity = Singularity::NonSingular;
+    }
+
+
+        //====================================================
+        // BEGIN Replacement solve with fixed prime
+
+    Ints::Element d;
+    Method::Dixon m(M);
+    typedef Givaro::Modular<double> Field;
+		// 0.7213475205 is an upper approximation of 1/(2log(2))
+    size_t bitsize((size_t)( 26-(int)ceil(log((double)A.rowdim())*0.7213475205)));
```

Comment:
Use `FieldTraits<Givaro::Modular<double>>::betBitSize(A.rowdim())` instead (defined in `field/field-traits.h`

Reply:
ok done.